app/javascript/components/provider-form/nested-provider.jsx

```diff
@@ -0,0 +1,29 @@
+import React, { useState, useContext } from 'react';
+
+import { EditingContext } from './index';
+import Select from '../select';
+import { useFormApi, validatorTypes } from '@@ddf';
+
+// Recursively omits the required validator and the isRequired attribute
+const override = fields => fields.map(({
```

Comment:
@agrare I'm starting to have this feeling that the `override` is a little bit of overkill here. If I imagine a user trying to edit a provider with a linked provider, I'd imagine that the linked credentials are pre-populated into the form (except the password but that's fine). At least that's the standard behavior right now, so I don't really understand why these creds can't be required. We could just hide the underlying fields with a simple condition if we're not in edit mode and there's a linked provider selected in the form.

Reply:
That's definitely fine, I'm just not familiar with how to do that (allowing edit but not create)